tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -125,6 +125,11 @@ public static IDatadogLogger GetLoggerFor<T>()
             return GetLoggerFor(typeof(T));
         }
 
+        internal static void CloseAndFlush()
+        {
+            Log.CloseAndFlush();
```

Comment:
This closes the static Serilog instance, _not_ our instance. Instead you want this
```suggestion
            SharedLogger.CloseAndFlush();
```

And then add this to `DatadogSerilogLogging` (make _logger non-readonly): 

```csharp
public void CloseAndFlush()
{
    var logger = Interlocked.Exchange(ref _logger, SilentLogger.Instance);

    (logger as IDisposable)?.Dispose();
}
```

And add this to `IDatadogLogger`:

```csharp
void CloseAndFlush();
```